### ISSUE TITLE
BUG: SHAP force plots HTML is broken (regression)

### ISSUE THREAD

**Issue by richarddli** [shap/shap#3448]
### Issue Description

In https://github.com/shap/shap/commit/95493ce93a90a86b0fb46ba2288bb86a08c4908e#diff-2c2e80cfb3ae153160970a29182c8943d1174f829138b2d944fc66479ccf6201, `.format` was switched to use f-strings. However, the code change replaced a single call to `id_generator()` with two calls. By design, `id_generator` returns a random ID each time.

The consequence of this change is that the IDs will not match, which means the force plot is never displayed. This issue is present in 0.44 but not in 0.43.

### Minimal Reproducible Example

```
        shap_values_single = explainer.shap_values(row, nsamples=NSHAP_SAMPLES)
        shap.force_plot(explainer.expected_value, shap_values_single, row).html()
```

Note that the IDs returned in the html do not match.

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [X] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

Master

**Comment by CloseChoice** [comment]
Good catch, thanks for reporting. I will have a look and fix this.

@connortann : Could we get a regression tag?

**Comment by connortann** [comment]
The fix is now in master and will be in the next hotfix release `0.44.1`, which will hopefully be out in a day or two.

**Comment by connortann** [comment]
This should be fixed in the latest release `0.44.1` which has been published to PyPI and will shortly be on conda-forge. Please let us know if you still encounter the issue.
